### DISCUSSION TITLE
multiple response models with status code

### DISCUSSION THREAD

**Discussion by DeoLeung** [fastapi/fastapi#8592] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi import Response
from pydantic import BaseModel

app = FastAPI()


class OK(BaseModel):
  a: str


class NotOK(BaseModel):
  b: str


@app.get('/a', response_model=OK, responses={400: {'model': NotOK}})
def no_exception(res: Response):
  """this fails"""
  res.status_code = 400
  return {'b': 'abc'}


@app.get('/b', responses={400: {'model': NotOK}, 200: {'model': OK}})
def no_exception2(res: Response):
  """this works"""
  res.status_code = 400
  return {'b': 'abc'}
```

### Description

when calling `/a`, we would expect it to use model `NotOK` to validate the response, but it uses model `OK`, which will raise exception

```bash
pydantic.error_wrappers.ValidationError: 1 validation error for OK
response -> a
  field required (type=value_error.missing)
```

if `responses` is used, shall be checked first, or we could update the [document](https://fastapi.tiangolo.com/advanced/additional-responses/#additional-response-with-model)

### Operating System

Linux

### FastAPI Version

0.73.0

### Python Version

3.9

**Comment by Kludex** [comment]
What's the problem here?

**Comment by DeoLeung** [comment]
> What's the problem here?

I updated the issue with more detailed description

**Comment by yinziyan1206** [comment]
return response directly
like:
    return JSONResponse(status_code=400, content={"b": "abc"})
or use exception_handler
